clang/lib/Sema/SemaChecking.cpp

```diff
@@ -1907,6 +1957,68 @@ Sema::CheckBuiltinFunctionCall(FunctionDecl *FDecl, unsigned BuiltinID,
       return ExprError();
     }
     break;
+
+  // We allow using CHERI builtins with any pointer type and (u)intcap_t.
+  // Note: The following are not polymorphic yet (and should be used so rarely
+  // that it probably doesn't matter):
+  // TODO: __builtin_cheri_cap_build
+  // TODO: __builtin_cheri_cap_type_copy
+  // TODO: __builtin_cheri_conditional_seal
+  // TODO: __builtin_cheri_cap_from_pointer
+  // TODO: __builtin_cheri_cap_to_pointer
```

Comment:
Hmm the inconsistency is a bit sad, and these shouldn't be any more difficult, just a couple more cases? Especially `__builtin_cheri_conditional_seal`, that's the same as `__builtin_cheri_seal`.

Reply:
However, I think the arguments are inverted. We should probably solve that inconsistency.